### ISSUE TITLE
Failed to extract eigenvector(s) from GRM.

### ISSUE THREAD

**Issue by bioinfornatics** [chrchang/plink-ng#135]
Dear,

I build plink 1.9 using rev 6798f82 
And I encounter this error:

```
$ plink --bfile "${d}_all_prun" \
            --read-freq "${d}.frqx" \
            --allow-no-sex \
            --within "${d}.cluster" \
            --mwithin 1 \
            --pca-clusters "${d}.cluster.superpop" \
            --pca ${pc} \
            --out "${file}_proj"

PLINK v1.90p 64-bit (2 Nov 2019)               www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to test_FinalReport_proj.log.
Options in effect:
  --allow-no-sex
  --bfile _all_prun
  --mwithin 1
  --out test_FinalReport_proj
  --pca 2
  --pca-clusters .cluster.superpop
  --read-freq .frqx
  --within .cluster

257738 MB RAM detected; reserving 128869 MB for main workspace.
Allocated 96651 MB successfully, after larger attempt(s) failed.
253 variants loaded from .bim file.
2511 people (3 males, 4 females, 2504 ambiguous) loaded from .fam.
Ambiguous sex IDs written to test_FinalReport_proj.nosex .
--within: 6 clusters loaded, covering a total of 2511 people.
Using up to 39 threads (change this with --threads).
Before main variant filters, 2508 founders and 3 nonfounders present.
Calculating allele frequencies... done.
--read-freq: .frqx file loaded.
253 variants and 2511 people pass filters and QC.
Note: No phenotypes present.
--pca-cluster-names/--pca-clusters: 2504 samples specified.
Relationship matrix calculation complete.
[extracting eigenvalues and eigenvectors]
Error: Failed to extract eigenvector(s) from GRM.
```

So I do not understand the reason to this faillure, indeed any provided makefile try to link plink to the Eigen library !

my buildscript

```bash
export OPT_FLAGS='-O3 -g -Wall -Wformat-security -Wp,-D_GLIBCXX_ASSERTIONS -fPIC -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables '"-mtune=${TUNE}"
export CXX=$(which g++)
export CC=$(which gcc)
export CFLAGS="${OPT_FLAGS} ${LAPACK_CFLAGS}"
export CXXFLAGS="${OPT_FLAGS}"
export ZLIB_LDFLAGS="-L/opt/zlib-original-1.2.11/lib -lz"
export ZLIB_CFLAGS="-I/opt/zlib-original-1.2.11/include"
export LAPACK_CFLAGS"-I/opt/lapack-3.9.0/include/cblas -I/opt/lapack-3.9.0/include/lapacke"
export LAPACK_LDFLAGS="-L/opt/lapack-3.9.0/lib64 -lcblas -lblas -llapacke -llapack"
export LDFLAGS+="-Wl,-z,relro -Wl,-z,now -Wl,--as-needed -lpthread ${LAPACK_LDFLAGS}"

cd plink-ng-master/1.9/
#mv -f Makefile.std Makefile
for f in pigz.c dose2plink.c plink_common.h; do
    sed -e 's,"zlib-1.2.8/zlib.h",<zlib.h>,g' -i "$f"
done
for f in *.c *.h; do
    sed -e 's/define PRId64 "lld"/define PRId64 "ld"/g;s/define PRIu64 "llu"/define PRIu64 "lu"/g' -i~ "$f"
done
gcc ${CFLAGS} -o prettify prettify.c
make CFLAGS="${CFLAGS} -DDYNAMIC_ZLIB -std=gnu++11 -Wno-long-long ${ZLIB_CFLAGS} -Wl,-rpath,${ZLIB_LIBDIR}:${C_ROOT}/lib64 -lpthread" \
     ZLIB="${ZLIB_LDFLAGS}" \
     NO_LAPACK=1 \
     plink
```

thanks for your help

**Comment by chrchang** [comment]
Can you try rerunning the PCA with plink 2.0?  It should provide a more informative error message.

**Comment by chrchang** [comment]
(Replace "--pca-clusters" with "--require-pheno CATPHENO".  This won't perform projection for the samples outside your clusters, but that's irrelevant for diagnostic purposes since your command is failing before that point.)

**Comment by bioinfornatics** [comment]
Thanks @chrchang for your help. Unfortunately, the production team is bound to release 1.90xxx and can not change their pipeline to the 2.xxx

**Comment by bioinfornatics** [comment]
The new plink major version does not mention the support of `ped` file format. 
This is an important issue to upgrade plink to version 2 and more

**Comment by chrchang** [comment]
I only asked to use plink 2.0 to see a more informative error message.

More generally, plink 2.0 alpha is not intended to be a standalone program; you’re *supposed* to keep using plink 1.9 for everything that isn’t implemented in 2.0 yet.

**Comment by bioinfornatics** [comment]
Thank @chrchang
